packages/frontend/src/utils/useDebounce.tsx

```diff
@@ -0,0 +1,23 @@
+import BigNumber from 'bignumber.js'
+import { useState, useEffect } from 'react'
+
+export default function useDebounce<T extends Number | BigNumber | String | Boolean>(value: T, delay: number) {
+  // State and setters for debounced value
+  const [debouncedValue, setDebouncedValue] = useState(value)
+  useEffect(
```

Comment:
I can see value can be a big number in the function definition. I think we should use useAppEffect.